### PR TITLE
move gcp machineset functionality to remotemachinesets controller

### PR DESCRIPTION
**Pull request by staebler**
https://jira.coreos.com/browse/CO-657

### CODE REVIEW

File: pkg/controller/remotemachineset/awsactuator.go [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+		return nil, fmt.Errorf("could not decode AWS ProviderConfig: %v", err)
+	}
+	spec, ok := obj.(*awsprovider.AWSMachineProviderConfig)
+	if !ok {
+		return nil, fmt.Errorf("Unexpected object: %#v", gvk)
+	}
+	return spec, nil
+}
+
+func encodeAWSMachineProviderSpec(awsProviderSpec *awsprovider.AWSMachineProviderConfig, scheme *runtime.Scheme) (*runtime.RawExtension, error) {
```
[the remark this answers is not in the document]
Comment (staebler) [reply pkg/controller/remotemachineset/awsactuator.go]: Yes.

File: pkg/controller/remotemachineset/awsactuator_test.go [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+			defer mockCtrl.Finish()
+
+			gClient := mockaws.NewMockClient(mockCtrl)
+
+			// set up mock expectations
+			if test.mockAWSClient != nil {
+				test.mockAWSClient(gClient)
+			}
+
+			ga := &AWSActuator{
```
[the remark this answers is not in the document]
Comment (staebler) [reply pkg/controller/remotemachineset/awsactuator_test.go]: Yes

File: pkg/controller/remotemachineset/remotemachineset_controller_test.go [hunk 3]
```diff
@@ -307,121 +260,74 @@ func TestRemoteMachineSetReconcile(t *testing.T) {
...
+				localExisting = append(localExisting, testSecret(adminKubeconfigSecret, adminKubeconfigSecretKey, testName))
+			}
+			fakeClient := fake.NewFakeClient(localExisting...)
 			remoteFakeClient := fake.NewFakeClient(test.remoteExisting...)
 
 			mockCtrl := gomock.NewController(t)
-			mockAWSClient := mockaws.NewMockClient(mockCtrl)
-			// Test availability zone retrieval when zones have not been set for machine pool
-			mockTestAvailabilityZones(mockAWSClient, "us-east-1", []string{"us-east-1a", "us-east-1b", "us-east-1c"})
+			mockActuator := mock.NewMockActuator(mockCtrl)
```
[the remark this answers is not in the document]
Comment (staebler) [reply pkg/controller/remotemachineset/remotemachineset_controller_test.go]: Yes

**Comment by joelddiaz** [comment]
@staebler i guess lets go ahead and squash, and i'll re-lgtm

**Comment by staebler** [comment]
Want to hold this for a successful e2e-gcp run.

/hold
/test e2e-gcp

**Comment by staebler** [comment]
@joelddiaz I pushed up a fix for the e2e gcp tests where the test for scaling the machinesets was looking for the wrong prefix for the machines. Please review 7309db0774851ac3f4ccfeea66fcc168c52daf16 at your leisure. I will squash once you are satisfied with the changes.

**Comment by joelddiaz** [comment]
> @joelddiaz I pushed up a fix for the e2e gcp tests where the test for scaling the machinesets was looking for the wrong prefix for the machines. Please review [7309db0](https://github.com/openshift/hive/commit/7309db0774851ac3f4ccfeea66fcc168c52daf16) at your leisure. I will squash once you are satisfied with the changes.

@staebler go ahead an squash

**Comment by staebler** [comment]
> > @joelddiaz I pushed up a fix for the e2e gcp tests where the test for scaling the machinesets was looking for the wrong prefix for the machines. Please review [7309db0](https://github.com/openshift/hive/commit/7309db0774851ac3f4ccfeea66fcc168c52daf16) at your leisure. I will squash once you are satisfied with the changes.
> 
> @staebler go ahead an squash

Squashed.

**Pull request merged**